### ISSUE TITLE
pubsub doesn't work if using  `withAuthenticator` from `@aws-amplify/ui-react`

### ISSUE THREAD

**Issue by smnodame** [aws-amplify/amplify-js#9765]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

PubSub

### Environment information

<details>

```
 System:
    OS: Linux 5.4 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 5.17 GB / 15.30 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chrome: 99.0.4844.84
    Firefox: 98.0.2
  npmPackages:
    @aws-amplify/ui-react: ^2.13.0 => 2.13.0 
    @aws-amplify/ui-react-internal:  undefined ()
    @aws-amplify/ui-react-legacy:  undefined ()
    @testing-library/jest-dom: ^5.11.4 => 5.14.1 
    @testing-library/react: ^11.1.0 => 11.2.7 
    @testing-library/user-event: ^12.1.10 => 12.8.3 
    aws-amplify: ^4.3.18 => 4.3.18 
    aws-amplify-react: ^5.0.14 => 5.1.9 
    bootstrap: ^5.1.1 => 5.1.1 
    lodash: ^4.17.21 => 4.17.21 
    react: ^17.0.2 => 17.0.2 
    react-bootstrap: ^1.6.3 => 1.6.3 
    react-bootstrap/AbstractNav:  undefined ()
    react-bootstrap/AbstractNavItem:  undefined ()
    react-bootstrap/Accordion:  undefined ()
    react-bootstrap/AccordionCollapse:  undefined ()
    react-bootstrap/AccordionContext:  undefined ()
    react-bootstrap/AccordionToggle:  undefined ()
    react-bootstrap/Alert:  undefined ()
    react-bootstrap/Badge:  undefined ()
    react-bootstrap/BootstrapModalManager:  undefined ()
    react-bootstrap/Breadcrumb:  undefined ()
    react-bootstrap/BreadcrumbItem:  undefined ()
    react-bootstrap/Button:  undefined ()
    react-bootstrap/ButtonGroup:  undefined ()
    react-bootstrap/ButtonToolbar:  undefined ()
    react-bootstrap/Card:  undefined ()
    react-bootstrap/CardColumns:  undefined ()
    react-bootstrap/CardContext:  undefined ()
    react-bootstrap/CardDeck:  undefined ()
    react-bootstrap/CardGroup:  undefined ()
    react-bootstrap/CardImg:  undefined ()
    react-bootstrap/Carousel:  undefined ()
    react-bootstrap/CarouselCaption:  undefined ()
    react-bootstrap/CarouselItem:  undefined ()
    react-bootstrap/CloseButton:  undefined ()
    react-bootstrap/Col:  undefined ()
    react-bootstrap/Collapse:  undefined ()
    react-bootstrap/Container:  undefined ()
    react-bootstrap/Dropdown:  undefined ()
    react-bootstrap/DropdownButton:  undefined ()
    react-bootstrap/DropdownItem:  undefined ()
    react-bootstrap/DropdownMenu:  undefined ()
    react-bootstrap/DropdownToggle:  undefined ()
    react-bootstrap/ElementChildren:  undefined ()
    react-bootstrap/Fade:  undefined ()
    react-bootstrap/Feedback:  undefined ()
    react-bootstrap/Figure:  undefined ()
    react-bootstrap/FigureCaption:  undefined ()
    react-bootstrap/FigureImage:  undefined ()
    react-bootstrap/Form:  undefined ()
    react-bootstrap/FormCheck:  undefined ()
    react-bootstrap/FormCheckInput:  undefined ()
    react-bootstrap/FormCheckLabel:  undefined ()
    react-bootstrap/FormContext:  undefined ()
    react-bootstrap/FormControl:  undefined ()
    react-bootstrap/FormFile:  undefined ()
    react-bootstrap/FormFileInput:  undefined ()
    react-bootstrap/FormFileLabel:  undefined ()
    react-bootstrap/FormGroup:  undefined ()
    react-bootstrap/FormLabel:  undefined ()
    react-bootstrap/FormText:  undefined ()
    react-bootstrap/Image:  undefined ()
    react-bootstrap/InputGroup:  undefined ()
    react-bootstrap/Jumbotron:  undefined ()
    react-bootstrap/ListGroup:  undefined ()
    react-bootstrap/ListGroupItem:  undefined ()
    react-bootstrap/Media:  undefined ()
    react-bootstrap/Modal:  undefined ()
    react-bootstrap/ModalBody:  undefined ()
    react-bootstrap/ModalContext:  undefined ()
    react-bootstrap/ModalDialog:  undefined ()
    react-bootstrap/ModalFooter:  undefined ()
    react-bootstrap/ModalHeader:  undefined ()
    react-bootstrap/ModalTitle:  undefined ()
    react-bootstrap/Nav:  undefined ()
    react-bootstrap/NavContext:  undefined ()
    react-bootstrap/NavDropdown:  undefined ()
    react-bootstrap/NavItem:  undefined ()
    react-bootstrap/NavLink:  undefined ()
    react-bootstrap/Navbar:  undefined ()
    react-bootstrap/NavbarBrand:  undefined ()
    react-bootstrap/NavbarCollapse:  undefined ()
    react-bootstrap/NavbarContext:  undefined ()
    react-bootstrap/NavbarToggle:  undefined ()
    react-bootstrap/Overlay:  undefined ()
    react-bootstrap/OverlayTrigger:  undefined ()
    react-bootstrap/PageItem:  undefined ()
    react-bootstrap/Pagination:  undefined ()
    react-bootstrap/Popover:  undefined ()
    react-bootstrap/PopoverContent:  undefined ()
    react-bootstrap/PopoverTitle:  undefined ()
    react-bootstrap/ProgressBar:  undefined ()
    react-bootstrap/ResponsiveEmbed:  undefined ()
    react-bootstrap/Row:  undefined ()
    react-bootstrap/SafeAnchor:  undefined ()
    react-bootstrap/SelectableContext:  undefined ()
    react-bootstrap/Spinner:  undefined ()
    react-bootstrap/SplitButton:  undefined ()
    react-bootstrap/Switch:  undefined ()
    react-bootstrap/Tab:  undefined ()
    react-bootstrap/TabContainer:  undefined ()
    react-bootstrap/TabContent:  undefined ()
    react-bootstrap/TabContext:  undefined ()
    react-bootstrap/TabPane:  undefined ()
    react-bootstrap/Table:  undefined ()
    react-bootstrap/Tabs:  undefined ()
    react-bootstrap/ThemeProvider:  undefined ()
    react-bootstrap/Toast:  undefined ()
    react-bootstrap/ToastBody:  undefined ()
    react-bootstrap/ToastContext:  undefined ()
    react-bootstrap/ToastHeader:  undefined ()
    react-bootstrap/ToggleButton:  undefined ()
    react-bootstrap/ToggleButtonGroup:  undefined ()
    react-bootstrap/Tooltip:  undefined ()
    react-bootstrap/createChainedFunction:  undefined ()
    react-bootstrap/createWithBsPrefix:  undefined ()
    react-bootstrap/divWithClassName:  undefined ()
    react-bootstrap/helpers:  undefined ()
    react-bootstrap/transitionEndListener:  undefined ()
    react-bootstrap/triggerBrowserReflow:  undefined ()
    react-bootstrap/types:  undefined ()
    react-bootstrap/usePopperMarginModifiers:  undefined ()
    react-bootstrap/useWrappedRefWithWarning:  undefined ()
    react-dom: ^17.0.2 => 17.0.2 
    react-scripts: 4.0.3 => 4.0.3 
    web-vitals: ^1.0.1 => 1.1.2 
  npmGlobalPackages:
    corepack: 0.10.0
    npm: 8.5.0



```

</details>

### Describe the bug

pubsub doesn't send any requests. it works when I use withAuthenticator from `aws-amplify-react`. I can see a ws request after click a button but after changing to use `withAuthenticator` from `@aws-amplify/ui-react`, it doesn't send any requests and no error message on the console

### Expected behavior

pubsub should send a reboot command to AWS and I should see it if I subscribe with #

### Reproduction steps

1. run the following code
2. log in
3. click the reboot button

### Code Snippet

```javascript
import Amplify from 'aws-amplify';
import { withAuthenticator } from '@aws-amplify/ui-react'
import awsconfig from './aws-exports';
import { PubSub } from 'aws-amplify';
import { AWSIoTProvider } from '@aws-amplify/pubsub/lib/Providers';

Amplify.configure({
  ...awsconfig
});

Amplify.addPluggable(new AWSIoTProvider({
  aws_pubsub_region: 'us-east-2',
  aws_pubsub_endpoint: 'wss://xxx-ats.iot.us-east-2.amazonaws.com/mqtt',
}));

const App = () => {
  const reboot = () => {
    PubSub.publish('cmd', {
      action: 'reboot'
    })
  }

  return <div>
    <button onClick={reboot}>reboot</button>
  </div>
}
export default withAuthenticator(App, true);

```

### Log output

there is no logs

### aws-exports.js

```
const awsmobile = {
    "aws_project_region": "us-east-2",
    "aws_cognito_identity_pool_id": "us-east-2:xxx-43fc-b299-3ef2ce046af2",
    "aws_cognito_region": "us-east-2",
    "aws_user_pools_id": "us-east-2_xxx",
    "aws_user_pools_web_client_id": "5ehosvumi17ffv3fs0e5xxxx",
    "oauth": {},
    "aws_cognito_login_mechanisms": [
        "EMAIL"
    ],
    "aws_cognito_signup_attributes": [
        "EMAIL"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [
        "SMS"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": []
    },
    "aws_cloud_logic_custom": [
        {
            "name": "user",
            "endpoint": "https://xxx.execute-api.us-east-2.amazonaws.com/staging",
            "region": "us-east-2"
        },
        {
            "name": "iot",
            "endpoint": "https://xxx.execute-api.us-east-2.amazonaws.com/staging",
            "region": "us-east-2"
        }
    ]
}
```

### Manual configuration

-

### Additional configuration

-

### Additional information and screenshots
I can see only ws requests for react

![image](https://user-images.githubusercontent.com/2629590/161521537-4b7e68da-b8c1-44ab-a269-04439ed7e783.png)

**Comment by stocaaro** [comment]
Hello @smnodame,

I made a run at reproducing your issue this afternoon. I built a new React sample app going through the [setup steps for Auth](https://docs.amplify.aws/lib/auth/getting-started/q/platform/js/#configure-your-application) and [for pubsub against IoT](https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/). I followed [the instructions](https://ui.docs.amplify.aws/components/authenticator) to use the current `withAuthenticator` UI component.

I ran into some Typescript compilation errors along the way, but in spite of these errors, I was able to send and receive messages from the IoT topic.

Since this was working for you prior to upgrading to the latest UI component, something about that upgrade process may have gone wrong.

Try turning the amplify log level up by adding the following line right after your `Amplify.configure` line.
```
Amplify.Logger.LOG_LEVEL = 'DEBUG';
```

It would be interesting to see what is and isn't working in your current broken version with the logging turned up.

I have investigated issues in the past where `aws-amplify-react` causes old versions of packages to be used. I would recommend making sure this package isn't in your `package.json` file, then I would recommend removing your `node_modules`, running `npm update` and then reinstalling everything `npm i`.

Another set of issues can be caused by mixing PubSub with API, which has dependencies on PubSub. The problem arises by using `import { PubSub } from 'aws-amplify';` where the API category is accessing PubSub category via `import { PubSub } from '@aws-amplify/pubsub';`. Having both of these used in the same runtime causes issues that can be resolved by accessing PubSub exclusively via `@aws-amplify/pubsub`.

Interested to get more information on what is happening here.  Hopefully something in this will be helpful!
